### PR TITLE
Preserve curly braces in encapsed strings

### LINKED ISSUE #6: Curlys in encapsed strings
Curlys in encapsed strings are not fully supported yet (`"two {$b}s"` vs. `"two ${b}s";`).

See https://github.com/glayzzle/php-parser/issues/101

### PR DESCRIPTION
**Pull request by czosel**
This is based on https://github.com/glayzzle/php-parser/pull/143.

Fixes #6, #210, #211

### CODE REVIEW

File: tests/string/__snapshots__/jsfmt.spec.js.snap [hunk 1]
```diff
@@ -36,12 +36,13 @@ $string = "foo\\x";
 $string = "foo\\u{ff}bar";
 $string = "foo\\ubar";
 $string = "foo\\u";
-$string = "He drank some $juice juice.";
-$string = "He drank some {$juice} juice.";
-$string = "He drank some \${juice} juice.";
-$string = "He drank some \${$juice} juice.";
-$string = "He drank some juice made of \${juice}s.";
-$string = "He drank some $juices[0] juice.";
+$string = "simple $encapsed";
```
Comment (evilebottnawi) [inline tests/string/__snapshots__/jsfmt.spec.js.snap]: Why difference text? :confused:

File: tests/string/string.php [hunk 2]
```diff
@@ -33,12 +33,13 @@
...
-$string = "He drank some ${$juice} juice.";
-$string = "He drank some juice made of ${juice}s.";
-$string = "He drank some $juices[0] juice.";
+$string = "simple $encapsed";
+$string = "simple $encapsed[0]";
+$string = "encapsed variable ${name}.";
+$string = "encapsed variable {$value}.";
+$string = "encapsed offset {$value[0]}.";
+$string = "encapsed {$method->call()}.";
+$string = "combined ${$encapsed}.";
```
Comment (czosel) [inline tests/string/string.php]: @evilebottnawi i had to change some of the texts for debugging - got some errors when running with `AST_COMPARE=1` that were hard to track down when all the strings were so similar 😉

File: package.json [hunk 3]
```diff
@@ -10,7 +10,7 @@
     "src"
   ],
   "dependencies": {
-    "php-parser": "glayzzle/php-parser#82233e9f93a7525ed884bba65fb0369de12b0ab4",
+    "php-parser": "czosel/php-parser#0590fd4c129ddb1298a8639e69f134afb5977d20",
```
Comment (evilebottnawi) [inline package.json]: Let's wait when `php-parser` merge you PR and we can merge this PR :+1:

Comment (czosel) [reply package.json]: :+1:

**Comment by mgrip** [comment]
@czosel updated with latest from php-parser after your PR was merged. One failing test now, going to start looking into it

**Comment by mgrip** [comment]
This will also fix https://github.com/prettier/plugin-php/issues/250 (locations for different node types where fixed in other parser updates)

**Comment by evilebottnawi** [comment]
@mgrip need investigate with comments :confused:

**Comment by czosel** [comment]
Maybe it would be best to merge this and deal with the broken comments in a separate PR?

**Comment by mgrip** [comment]
works for me! I think it got introduced with some other changes in the parser before your change got merged into it (definitely still possible its a bug on our end though)

**Pull request merged**